src/worldstate/src/WorldStateNode.cpp

```diff
@@ -0,0 +1,141 @@
+//         * Left -> k = 1;  Right -> k = 2;  Top -> k = 3
+//         * int   G   = number of gate bars seen
+//         * float D_k = distance from camera to bar k
+//         * bool  g_k = bar k visible
+//         */
+//
+//        /* State! G = 0
+//         * output is "locatingGate" unless previous state was passingGate,
+//         * in which case the output is "passingGate." This is under the
+//         assumption
+//         * that the robot is moving through the gate and the bars have simply
+//         * disappeared from view
+//         */
+//
+//        /* State! G = 1
+//         * output is "aligningWithGate" A pole has been found and the robot
+//         should
+//         * take the appropriate maneuvers to align with the gate
+//         appropriately.
+//         */
+//
+//        /* State! G = 2 or 3
+//         * - If g_1, g_2, and g_3, and |D_1 - D_2| < error and D_3 >=
+//         clearance then "passingGate"
+//         * - If !g_1, !g_2, and !g_3, go back to "locatingGate"
+//         * - Otherwise "aligningWithGate"
+//         */
+//
+//        case locatingGate:
+//            //If any of the poles has been found
+//            if (msg->detectLeft || msg->detectRight || msg->detectTop) {
+//                current_state_ = aligningWithGate;
+//                //If all three poles are in view
+//                if (msg->detectLeft && msg->detectRight) {
+//                    // and the robot is relatively aligned
+//                    if (distBtwnHorizontalGatePole < errorTolerance &&
+//                    msg->distanceTop >= clearanceHeight)
+//                        current_state_ = passingGate;
+//                }
+//            }
+//            break;
+//
+//        case aligningWithGate:
+//            //If the poles suddenly disappear from view
+//            if (!msg->detectTop && !msg->detectLeft && !msg->detectRight){
+//                current_state_ = locatingGate;
+//            }
+//
+//            //Assuming all poles are in view
+//            if (msg->detectTop && msg->detectLeft && msg->detectRight){
+//                // and the robot is relatively aligned
+//                if (distBtwnHorizontalGatePole < errorTolerance &&
+//                msg->distanceTop >= clearanceHeight)
+//                    current_state_ = passingGate;
+//            }
+//            break;
+//
+//        case passingGate:
+//            /* Handle Error Exceptions */
+//
+//
+//            //If the poles suddenly disappear from view, assume the robot has
+//            bee-lined and
+//            //passed through the gate successfully
+//            if (!msg->detectTop && !msg->detectLeft && !msg->detectRight){
+//                current_state_ = locatingPole;
+//            }
+//            break;
+//
+//        default:
+//            break;
+//    }
+//
+//    gateDetMsg.state = current_state_;
+//
+//    world_state_publisher_.publish(gateDetMsg);
+//}
```

Comment:
Not sure if the intent was to merge this in, but we should uncomment it if so, or remove it if not.

Reply:
Removed the comments!